### DISCUSSION TITLE
Gatsby Starter

### DISCUSSION THREAD

**Discussion by peiche** [codyhouse/codyhouse-framework#77] [Show and tell]
I've created a Gatsby starter using CodyFrame:

### [Gatsby + CodyHouse Starter](https://github.com/peiche/gatsby-starter-codyhouse)

[Components](https://codyhouse.co/ds/components) can be added to the Gatsby site within the `src/components` directory. The SCSS code should be placed within a new include created inside the `src/styles/components` directory (and don't forget to update the `_index.scss` file in the same directory). The JS code should be placed within its own file in `static/js` alongside `util.js`. Follow the CodyHouse [directions](https://codyhouse.co/blog/post/using-the-codyhouse-components-with-react) for creating/removing the script element in the component's componentDidMount and componentWillUnmount functions, respectively.

Support for using PurgeCSS is included but disabled by default. The config must be uncommented within `gatsby-config.json` and modified to fit your specific use-case(s).

**Comment by claudia-romano** [comment]
Hi Paul, thanks for the starter. We have included it in our Docs page:
https://codyhouse.co/ds/docs/framework#gatsby

Cheers!

**Comment by digiturnal** [comment]
The Starter while Deploying on Netlify and Vercel it fails. Same on Local

**Reply by digiturnal** [reply]
[Netlify Error Log]
12:31:30 PM: Build ready to start
12:31:32 PM: build-image version: 3bcb38c35508b42e9121d4badfe6d8c66fd7a3f0
12:31:32 PM: build-image tag: v4.3.2
12:31:32 PM: buildbot version: 04fb57a6b52e36fc0fd306c97e17aa22ebf93b50
12:31:32 PM: Fetching cached dependencies
12:31:32 PM: Failed to fetch cache, continuing with build
12:31:32 PM: Starting to prepare the repo for build
12:31:32 PM: No cached dependencies found. Cloning fresh repo
12:31:32 PM: git clone https://github.com/peiche/gatsby-starter-codyhouse
12:31:36 PM: Preparing Git Reference refs/heads/master
12:31:37 PM: Parsing package.json dependencies
12:31:38 PM: Starting build script
12:31:38 PM: Installing dependencies
12:31:38 PM: Python version set to 2.7
12:31:39 PM: Downloading and installing node v16.11.1...
12:31:39 PM: Downloading https://nodejs.org/dist/v16.11.1/node-v16.11.1-linux-x64.tar.xz...
12:31:39 PM: Computing checksum with sha256sum
12:31:39 PM: Checksums matched!
12:31:43 PM: Now using node v16.11.1 (npm v8.0.0)
12:31:43 PM: Started restoring cached build plugins
12:31:43 PM: Finished restoring cached build plugins
12:31:43 PM: Attempting ruby version 2.7.2, read from environment
12:31:44 PM: Using ruby version 2.7.2
12:31:45 PM: Using PHP version 8.0
12:31:45 PM: Started restoring cached node modules
12:31:45 PM: Finished restoring cached node modules
12:31:45 PM: Installing NPM modules using NPM version 8.0.0
12:32:04 PM: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
12:32:04 PM: npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
12:32:17 PM: npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
12:32:19 PM: npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
12:32:19 PM: npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
12:32:20 PM: npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
12:32:22 PM: npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
12:32:22 PM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
12:32:26 PM: npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
12:32:35 PM: npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
12:32:54 PM: added 2390 packages, and audited 2391 packages in 1m
12:32:54 PM: 230 packages are looking for funding
12:32:54 PM:   run `npm fund` for details
12:32:55 PM: 62 vulnerabilities (18 moderate, 42 high, 2 critical)
12:32:55 PM: To address issues that do not require attention, run:
12:32:55 PM:   npm audit fix
12:32:55 PM: To address all issues (including breaking changes), run:
12:32:55 PM:   npm audit fix --force
12:32:55 PM: Run `npm audit` for details.
12:32:55 PM: NPM modules installed
12:32:55 PM: Started restoring cached go cache
12:32:55 PM: Finished restoring cached go cache
12:32:55 PM: go version go1.16.5 linux/amd64
12:32:55 PM: go version go1.16.5 linux/amd64
12:32:55 PM: Installing missing commands
12:32:55 PM: Verify run directory
12:32:57 PM: ​
12:32:57 PM: ────────────────────────────────────────────────────────────────
12:32:57 PM:   Netlify Build                                                 
12:32:57 PM: ────────────────────────────────────────────────────────────────
12:32:57 PM: ​
12:32:57 PM: ❯ Version
12:32:57 PM:   @netlify/build 18.17.0
12:32:57 PM: ​
12:32:57 PM: ❯ Flags
12:32:57 PM:   baseRelDir: true
12:32:57 PM:   buildId: 6167f8f21e9dfab29d67862c
12:32:57 PM:   deployId: 6167f8f21e9dfab29d67862e
12:32:57 PM: ​
12:32:57 PM: ❯ Current directory
12:32:57 PM:   /opt/build/repo
12:32:57 PM: ​
12:32:57 PM: ❯ Config file
12:32:57 PM:   No config file was defined: using default values.
12:32:57 PM: ​
12:32:57 PM: ❯ Context
12:32:57 PM:   production
12:32:57 PM: ​
12:32:57 PM: ❯ Installing plugins
12:32:57 PM:    - @netlify/plugin-gatsby@1.0.3
12:33:05 PM: ​
12:33:05 PM: ❯ Loading plugins
12:33:05 PM:    - @netlify/plugin-gatsby@1.0.3 from Netlify app
12:33:06 PM: ​
12:33:06 PM: ────────────────────────────────────────────────────────────────
12:33:06 PM:   1. onPreBuild step from @netlify/plugin-gatsby                
12:33:06 PM: ────────────────────────────────────────────────────────────────
12:33:06 PM: ​
12:33:06 PM: No Gatsby cache found. Building fresh.
12:33:08 PM: Add `gatsby-plugin-netlify` to `gatsby-config.js` if you would like to support Gatsby redirects. 🎉
12:33:08 PM: ​
12:33:08 PM: (@netlify/plugin-gatsby onPreBuild completed in 2.1s)
12:33:08 PM: ​
12:33:08 PM: ────────────────────────────────────────────────────────────────
12:33:08 PM:   2. Build command from Netlify app                             
12:33:08 PM: ────────────────────────────────────────────────────────────────
12:33:08 PM: ​
12:33:08 PM: $ npm run build
12:33:08 PM: > gatsby-starter-codyhouse@1.0.0 build
12:33:08 PM: > gatsby build
12:33:14 PM: success open and validate gatsby-configs, load plugins - 2.547s
12:33:14 PM: success onPreInit - 0.005s
12:33:14 PM: success initialize cache - 0.062s
12:33:14 PM: success copy gatsby files - 0.087s
12:33:14 PM: success Compiling Gatsby Functions - 0.351s
12:33:14 PM: success onPreBootstrap - 0.366s
12:33:14 PM: success createSchemaCustomization - 0.029s
12:33:14 PM: success Checking for changed pages - 0.000s
12:33:14 PM: success source and transform nodes - 0.192s
12:33:16 PM: success building schema - 1.930s
12:33:16 PM: success createPages - 0.000s
12:33:17 PM: success createPagesStatefully - 0.259s
12:33:17 PM: info Total nodes: 43, SitePage nodes: 5 (use --verbose for breakdown)
12:33:17 PM: success Checking for changed pages - 0.000s
12:33:17 PM: success update schema - 0.402s
12:33:17 PM: success onPreExtractQueries - 0.000s
12:33:22 PM: success extract queries from components - 5.138s
12:33:22 PM: success write out redirect data - 0.004s
12:33:23 PM: success Build manifest and related icons - 1.150s
12:33:23 PM: success onPostBootstrap - 1.153s
12:33:23 PM: info bootstrap finished - 15.135s
12:33:23 PM: success run static queries - 0.016s - 2/2 123.27/s
12:33:23 PM: success run page queries - 0.013s - 5/5 398.13/s
12:33:23 PM: success write out requires - 0.029s
12:33:41 PM: failed Building production JavaScript and CSS bundles - 17.761s
12:33:41 PM: error Generating JavaScript bundles failed
12:33:41 PM: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
12:33:41 PM: ParserError: Syntax Error at line: 1, column 25
12:33:41 PM:     at /opt/build/repo/src/styles/main.scss:1:179430
12:33:41 PM:     at Parser.error (/opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:127:11)
12:33:41 PM:     at Parser.operator (/opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:162:20)
12:33:41 PM:     at Parser.parseTokens (/opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:245:14)
12:33:41 PM:     at Parser.loop (/opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:132:12)
12:33:41 PM:     at Parser.parse (/opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:51:17)
12:33:41 PM:     at parse (/opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-custom-properties/src/lib/postcss-values-parser.js:4:29)
12:33:41 PM:     at /opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-custom-properties/src/lib/transform-properties.js:11:21
12:33:41 PM:     at /opt/build/repo/node_modules/postcss/lib/container.js:72:18
12:33:41 PM:     at /opt/build/repo/node_modules/postcss/lib/container.js:55:18
12:33:41 PM:     at Rule.each (/opt/build/repo/node_modules/postcss/lib/container.js:41:16)
12:33:41 PM:     at Rule.walk (/opt/build/repo/node_modules/postcss/lib/container.js:52:17)
12:33:41 PM:     at /opt/build/repo/node_modules/postcss/lib/container.js:60:24
12:33:41 PM:     at Root.each (/opt/build/repo/node_modules/postcss/lib/container.js:41:16)
12:33:41 PM:     at Root.walk (/opt/build/repo/node_modules/postcss/lib/container.js:52:17)
12:33:41 PM:     at Root.walkDecls (/opt/build/repo/node_modules/postcss/lib/container.js:70:19)
12:33:41 PM:     at transformProperties (/opt/build/repo/node_modules/postcss-preset-env/node_modules/postcss-custom-properties/src/lib/transform-properties.js:8:7)
12:33:41 PM: ​
12:33:41 PM: ────────────────────────────────────────────────────────────────
12:33:41 PM:   "build.command" failed                                        
12:33:41 PM: ────────────────────────────────────────────────────────────────
12:33:41 PM: ​
12:33:41 PM:   Error message
12:33:41 PM:   Command failed with exit code 1: npm run build
12:33:41 PM: ​
12:33:41 PM:   Error location
12:33:41 PM:   In Build command from Netlify app:
12:33:41 PM:   npm run build
12:33:41 PM: ​
12:33:41 PM:   Resolved config
12:33:41 PM:   build:
12:33:41 PM:     command: npm run build
12:33:41 PM:     commandOrigin: ui
12:33:41 PM:     publish: /opt/build/repo/public
12:33:41 PM:     publishOrigin: ui
12:33:41 PM:   plugins:
12:33:41 PM:     - inputs: {}
12:33:41 PM:       origin: ui
12:33:41 PM:       package: '@netlify/plugin-gatsby'
12:33:42 PM: Caching artifacts
12:33:42 PM: Started saving node modules
12:33:42 PM: Finished saving node modules
12:33:42 PM: Started saving build plugins
12:33:42 PM: Finished saving build plugins
12:33:42 PM: Started saving pip cache
12:33:42 PM: Finished saving pip cache
12:33:42 PM: Started saving emacs cask dependencies
12:33:42 PM: Finished saving emacs cask dependencies
12:33:42 PM: Started saving maven dependencies
12:33:42 PM: Finished saving maven dependencies
12:33:42 PM: Started saving boot dependencies
12:33:42 PM: Finished saving boot dependencies
12:33:42 PM: Started saving rust rustup cache
12:33:42 PM: Finished saving rust rustup cache
12:33:42 PM: Started saving go dependencies
12:33:42 PM: Finished saving go dependencies
12:33:46 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
12:33:46 PM: Creating deploy upload records
12:33:46 PM: Failing build: Failed to build site
12:33:46 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2
12:33:46 PM: Finished processing build request in 2m14.644200041s

**Reply by digiturnal** [reply]
[Vercel Error Log]
12:36:12.263
Detected package.json
12:36:12.264
Installing dependencies...
12:36:12.278
Detected `package-lock.json` generated by npm 7...
12:36:30.824
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
12:36:31.954
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
12:36:41.027
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
12:36:42.241
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
12:36:42.359
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
12:36:42.874
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
12:36:44.351
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
12:36:46.140
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
12:36:48.432
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
12:37:00.104
npm WARN deprecated core-js@2.6.12: core-js@<3.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
12:37:08.830
12:37:08.830
added 2390 packages in 56s
12:37:08.830
12:37:08.830
230 packages are looking for funding
12:37:08.830
  run `npm fund` for details
12:37:09.135
Running "npm run build"
12:37:09.497
12:37:09.497
> gatsby-starter-codyhouse@1.0.0 build
12:37:09.497
> gatsby build
12:37:09.498
12:37:14.143
success open and validate gatsby-configs, load plugins - 2.167s
12:37:14.150
success onPreInit - 0.005s
12:37:14.200
success initialize cache - 0.050s
12:37:14.269
success copy gatsby files - 0.068s
12:37:14.580
success Compiling Gatsby Functions - 0.301s
12:37:14.582
success onPreBootstrap - 0.313s
12:37:14.622
success createSchemaCustomization - 0.004s
12:37:14.684
success Checking for changed pages - 0.000s
12:37:14.684
success source and transform nodes - 0.061s
12:37:15.389
success building schema - 0.705s
12:37:15.391
success createPages - 0.000s
12:37:15.474
success createPagesStatefully - 0.082s
12:37:15.474
info Total nodes: 43, SitePage nodes: 5 (use --verbose for breakdown)
12:37:15.475
success Checking for changed pages - 0.000s
12:37:15.624
success update schema - 0.148s
12:37:15.625
success onPreExtractQueries - 0.000s
12:37:17.925
success extract queries from components - 2.299s
12:37:17.929
success write out redirect data - 0.003s
12:37:18.191
success Build manifest and related icons - 0.258s
12:37:18.191
success onPostBootstrap - 0.261s
12:37:18.192
info bootstrap finished - 8.685s
12:37:18.217
success run static queries - 0.020s - 2/2 99.31/s
12:37:18.226
success run page queries - 0.009s - 5/5 573.92/s
12:37:18.234
success write out requires - 0.007s
12:37:32.138
failed Building production JavaScript and CSS bundles - 13.900s
12:37:32.152
error Generating JavaScript bundles failed
12:37:32.152
12:37:32.152
Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
12:37:32.152
ParserError: Syntax Error at line: 1, column 25
12:37:32.152
    at /vercel/path0/src/styles/main.scss:1:179430
12:37:32.152
    at Parser.error (/vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:127:11)
12:37:32.152
    at Parser.operator (/vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:162:20)
12:37:32.153
    at Parser.parseTokens (/vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:245:14)
12:37:32.153
    at Parser.loop (/vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:132:12)
12:37:32.153
    at Parser.parse (/vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-values-parser/lib/parser.js:51:17)
12:37:32.153
    at parse (/vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-custom-properties/src/lib/postcss-values-parser.js:4:29)
12:37:32.153
    at /vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-custom-properties/src/lib/transform-properties.js:11:21
12:37:32.153
    at /vercel/path0/node_modules/postcss/lib/container.js:72:18
12:37:32.153
    at /vercel/path0/node_modules/postcss/lib/container.js:55:18
12:37:32.153
    at Rule.each (/vercel/path0/node_modules/postcss/lib/container.js:41:16)
12:37:32.154
    at Rule.walk (/vercel/path0/node_modules/postcss/lib/container.js:52:17)
12:37:32.154
    at /vercel/path0/node_modules/postcss/lib/container.js:60:24
12:37:32.154
    at Root.each (/vercel/path0/node_modules/postcss/lib/container.js:41:16)
12:37:32.154
    at Root.walk (/vercel/path0/node_modules/postcss/lib/container.js:52:17)
12:37:32.154
    at Root.walkDecls (/vercel/path0/node_modules/postcss/lib/container.js:70:19)
12:37:32.154
    at transformProperties (/vercel/path0/node_modules/postcss-preset-env/node_modules/postcss-custom-properties/src/lib/transform-properties.js:8:7)
12:37:33.452
Error: Command "npm run build" exited with 1

**Reply by peiche** [reply]
Hi, I'm aware of this. I'm working on fixing it.

**Reply by peiche** [reply]
Hi @digiturnal, I've fixed the build problems and I can confirm that my [Netlify demo](https://gatsby-starter-codyhouse-demo.netlify.app/) is building again. Since CodyFrame 3 removed the fallback support, I've removed `postcss-preset-env` and `postcss-css-variables` from the build process. I added `autoprefixer` instead (and kept `postcss-calc`).
